marathon_acme/service.py

```diff
@@ -89,25 +92,37 @@ def listen_events(self, reconnects=0):
         self.log.info('Listening for events from Marathon...')
         self._attached = False
 
-        def on_finished(result, reconnects):
-            # If the callback fires then the HTTP request to the event stream
-            # went fine, but the persistent connection for the SSE stream was
-            # dropped. Just reconnect for now- if we can't actually connect
-            # then the errback will fire rather.
-            self.log.warn('Connection lost listening for events, '
-                          'reconnecting... ({reconnects} so far)',
-                          reconnects=reconnects)
+        def cb_listen(result, reconnects):
+            failure = None
+            if isinstance(result, Failure):
```

Comment:
I don't know how to test the non-Failure case. Calling `request.loseConnection()` results in a ResponseFailed failure, but in production I know that that does not happen, we get that PotentialDataLoss error instead. 😕